### PR TITLE
Java: Add Callable support for CodeDuplication.qll

### PR DESCRIPTION
**Pull request by Marcono1234**
Adds support for `Callable` for `CodeDuplication.qll`. This allows checking how similar constructors are, which can be useful for determining whether constructors should delegate to each other.

Maybe it would also be good to adjust the other predicates of the module, but I am not very familiar with that module.

### CODE REVIEW

File: java/ql/src/external/CodeDuplication.qll [hunk 1]
```diff
@@ -75,32 +75,45 @@ private predicate blockCoversStatement(int equivClass, int first, int last, Stmt
...
+
+/**
+ * Holds if the callables are non-empty and their bodies are identical.
  */
-predicate duplicateMethod(Method m, Method other) {
-  exists(int total | duplicateStatements(m, other, total, total))
+predicate duplicateCallable(Callable c, Callable other) {
+  exists(int total |
+    // Check count of `other`, otherwise statements of `c` might be a subset of `other`
+    duplicateStatements(c, other, total, total) and total = strictcount(statementInCallable(other))
```
Comment (Marcono1234) [inline java/ql/src/external/CodeDuplication.qll:115]: Checking for `total = strictcount(statementInCallable(other))` is a behavior change, but it looks like previously the predicate would have held even when `m` only had a subset of the statements of `other`.

**Comment by aschackmull** [comment]
The code duplication queries rely on duplication information being provided at extraction-time, and I believe we've stopped doing that by default. As a result, all the code duplication queries are currently marked as deprecated and won't yield any results, so updating this piece of code doesn't achieve much.

**Comment by Marcono1234** [comment]
Thanks for the information. Are there any plans to provide such information again in the future?

**Comment by aschackmull** [comment]
> Are there any plans to provide such information again in the future?

Not at the moment, no. We try to mostly focus on the parts of analysis that's relevant for security.

**Pull request closed without merging**